### PR TITLE
Fix order of  resending  messages after restart

### PR DESCRIPTION
**Pull request by alexggh**
The way we build the messages we need to send to approval-distribution can result in a situation where is we have multiple assignments covered by a coalesced approval, the messages are sent in this order:

ASSIGNMENT1, APPROVAL, ASSIGNMENT2, because we iterate over each candidate and add to the queue of messages both the assignment and the approval for that candidate, and when the approval reaches the approval-distribution subsystem it won't be imported and gossiped because one of the assignment for it is not known.

So in a network where a lot of nodes are restarting at the same time we could end up in a situation where a set of the nodes correctly received the assignments and approvals before the restart and approve their blocks and don't trigger their assignments. The other set of nodes should receive the assignments and approvals after the restart, but because the approvals never get broacasted anymore because of this bug, the only way they could approve is if other nodes start broadcasting their assignments.

I think this bug contribute to the reason the network did not recovered on `25-11-25 15:55:40` after the restarts.

Tested this scenario with a `zombienet` where `nodes` are finalising blocks because of aggression and all nodes are restarted at once and confirmed the network lags and doesn't recover before and it does after the fix

### CODE REVIEW

File: polkadot/node/core/approval-voting/src/lib.rs [hunk 1]
```diff
@@ -1864,6 +1866,7 @@ async fn distribution_messages_for_activation<Sender: SubsystemSender<RuntimeApi
 	}
 
 	messages[0] = ApprovalDistributionMessage::NewBlocks(approval_meta);
+	messages.extend(approvals.into_iter());
```
Comment (sandreim) [inline polkadot/node/core/approval-voting/src/lib.rs:1869]: I'd also add a comment here to explain why we send all approvals after assignments.

File: polkadot/node/core/approval-voting/src/lib.rs [hunk 2]
```diff
@@ -1582,8 +1582,9 @@ async fn handle_actions<
 					session_info_provider,
 				)
 				.await?;
-
-				approval_voting_sender.send_messages(messages.into_iter()).await;
+				for message in messages.into_iter() {
+					approval_voting_sender.send_unbounded_message(message);
```
Comment (sandreim) [inline polkadot/node/core/approval-voting/src/lib.rs:1586]: What is the reasoning to switch to unbounded ?

File: polkadot/node/core/approval-voting/src/lib.rs (hunk 1 above)
Comment (alexggh) [reply polkadot/node/core/approval-voting/src/lib.rs:1872]: done.

**Review by AndreiEres — APPROVED** [review]
Nice!

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/2025-11-25-kusama-parachains-spammening-aftermath/11108/1

**Pull request merged**